app/mender.go

```diff
@@ -682,19 +686,39 @@ func (m *Mender) InventoryRefresh() error {
 		{Name: "mender_client_version", Value: conf.VersionString()},
 	}
 
-	if idata == nil {
-		idata = make(client.InventoryData, 0, len(reqAttr))
+	if iData == nil {
+		iData = make(client.InventoryData, 0, len(reqAttr))
 	}
-	_ = idata.ReplaceAttributes(reqAttr)
+	_ = iData.ReplaceAttributes(reqAttr)
 
-	if idata == nil {
-		log.Infof("No inventory data to submit")
+	if iData == nil {
+		log.Info("No inventory data to submit")
 		return nil
 	}
 
-	err = ic.Submit(m.api.Request(m.authToken, nextServerIterator(m), reauthorize(m)), m.Config.Servers[0].ServerURL, idata)
+	// sort slice to have a reproducible hash
+	sort.Slice(iData, func(i, j int) bool { return strings.Compare(iData[i].Name, iData[j].Name) < 0 })
+
+	iDataBuf, err := json.Marshal(iData)
+	if err != nil {
+		return err
+	}
+	iDataSha := sha256.Sum256(iDataBuf)
+	lastInvSha, err := m.GetCurrentInventoryHash()
+	if err == nil && iDataSha == lastInvSha {
+		log.Info("The inventory data has not changed since the last time it was collected. Skipping server inventory update call")
+		return nil
+	}
+
+	log.Debugf("New inventory data, updating database with hash: %064x, inventory: %s", iDataSha, iData)
+	err = m.Store.WriteAll(datastore.InventoryHashKey, iDataSha[:])
```

Comment:
This should be done after `ic.Submit`, as pointed today in Sprint Review by @pasinskim

Reply:
Fixed now

Reply:
See https://github.com/mendersoftware/mender/compare/822e0fc90fa98f45e37eecb32dee62d7eaafca3b..2a77a52b5ebd787a73788d5b68f55e1663283669